src/test/java/javaslang/collection/euler/ProjectEulerTest.java

```diff
@@ -85,9 +84,37 @@ private static long fibonacci(int order) {
      * <p>See also <a href="https://projecteuler.net/problem=7">projecteuler.net problem 7</a>.</p>
      */
     @Test
-    @Ignore
-    public void shouldSolveProblem7() {
-        final int actual = Primes.asStream().drop(10_000).head();
-        assertThat(actual).isEqualTo(0); // TODO, see Primes#sieve(Stream)
+    public void test7() {
+        assertThat(primeNo(1)).isEqualTo(2);
+        assertThat(primeNo(2)).isEqualTo(3);
+        assertThat(primeNo(3)).isEqualTo(5);
+        assertThat(primeNo(4)).isEqualTo(7);
+        assertThat(primeNo(5)).isEqualTo(11);
+        assertThat(primeNo(6)).isEqualTo(13);
+        assertThat(primeNo(10_001)).isEqualTo(104_743);
+    }
+
+    private static int primeNo(int index) {
+        return Match
+                .when(1, (i) -> 2)
+                .orElse(() -> knownPrimes.drop(index - 1).head())
+                .apply(index);
+    }
+
+    private static final Stream<Integer> knownPrimes = Stream.gen(2, (p) -> nextPrime(p));
+
+    private static int nextPrime(int previousPrime) {
+        final Integer nextPrime = Match
+                .when(2, (i) -> 3)
+                .orElse(() -> Stream.gen(previousPrime + 2, (v) -> v + 2)
+                        .filter((i) -> !isEvenlyDiversableByKnownPrimes(previousPrime, i))
+                        .take(1).head())
+                .apply(previousPrime);
+        return nextPrime;
+    }
+
+    private static boolean isEvenlyDiversableByKnownPrimes(int previousPrime, int val) {
+        return knownPrimes.takeWhile((p) -> p < previousPrime).append(previousPrime)
```

Comment:
No, using `p <= previousPrime` doesn't work since `Stream.takeWhile(Predicate)` runs until it finds an element that doesn't satisfy the Predicate and since we here are inside the functions that builds up the tail of the `knownPrimes` stream `nextPrime(...)` will be called again and we will get a StackOverflow.